src/index.js

```diff
@@ -1,40 +1,104 @@
 import './style.css';
+import renderContent from './module/render.js';
+import { getLocalStorage, setLocalStorage } from './module/localStorage.js';
 
-const container = document.querySelector('.todo-listing');
-
-const todoList = [
-  {
-    description: 'Read my books',
-    completed: true,
-    index: 3,
-  },
-  {
-    description: 'Study data structures',
-    completed: true,
-    index: 1,
-  },
-  {
-    description: 'Learn new coding concepts',
-    completed: false,
-    index: 2,
-  },
-
-];
-
-const renderTask = () => {
-  const sortedList = todoList.sort((item1, item2) => item1.index - item2.index);
-
-  for (let i = 0; i < sortedList.length; i += 1) {
-    container.innerHTML += `
-    <div class="task-container">
-      <form class="task-form">
-        <input type="checkbox" class="checkbox">
-      </form>
-    <p class="task-description">${sortedList[i].description}</p>
-  </div>
-  <hr>
-    `;
+// const ulContainer = document.querySelector('.list-content');
+const inputData = document.querySelector('.todo-input');
+// const btnElement = document.querySelector('.btn');
+const mainContainer = document.querySelector('.wrapper');
+
+const todoList = [];
+
+const createTask = () => {
+  const taskObject = {};
+  taskObject.description = inputData.value;
+  taskObject.completed = false;
+  taskObject.index = todoList.length;
+  inputData.value = '';
+  return taskObject;
+};
+
+mainContainer.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    if (inputData.value !== '') {
+      const taskObject = createTask();
+      renderContent(taskObject);
+      todoList.push(taskObject);
+      todoList.sort((a, b) => a.index - b.index);
+      setLocalStorage(todoList);
+      getLocalStorage();
+      e.preventDefault();
+    }
+
+    if (e.target.className === 'task-description') {
+      const element = e.target.parentElement.parentElement.children[1].children[1].id;
+      todoList.forEach((elem) => {
+        if (elem.index === Number(element)) {
+          elem.description = e.target.value;
+        }
+      });
+      setLocalStorage(todoList);
+    }
+  }
+});
```

Comment:
- [ ] Your app seems to contain a few bugs. If I add a number of tasks and refresh the page, and add another task the old tasks are cleared from the local storage and only the new one is in the storage, although they still appear on the page, so when I refresh the page, the older tasks are gone. Kindly look into that.

- [ ] When a task is added, its `index` attribute should be set to be equal to the position in the array. So the first task should have index attributes of 1, the second one 2 and so on. Currently your tasks start at 0, instead of 1.